### PR TITLE
Support custom identifier for person name

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

As suggested by @razor-x on Slack, it's quite common to not have the email of an end-user available in PostHog (e.g. OSS projects who don't capture any PII). With this PR we now fall back to find a `name` or `username` to allow for different identifiers to be shown for a user.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by mariusandra** [comment]
Looks good. However I think we also show this email in the events table? 
image

**Review by Twixes — COMMENTED** [review]
This is great, though _I think_ Marius is right, this may not be the only place.

**Comment by paolodamico** [comment]
Right! Looking at this, we would ideally use the same component as we do for persons, as it also shows if a person is identified or not. Not very deep on how we fetch events from the DB, but would it be okay to replace https://github.com/PostHog/posthog/blob/27f271725a73340dad21e02df8a72896a773b6a4/posthog/api/event.py#L63 to send the email, name, username and is_identified props? Or would it cause too much load on the DB?

**Comment by Twixes** [comment]
I think it should be fine @paolodamico.

**Comment by paolodamico** [comment]
Awesome! Ready for another review, we now send the relevant properties from the event and show the same component in the events table.

**Comment by paolodamico** [comment]
Update: This PR ended up fixing an unintended behavior in which we would send the event's distinct ID as the person's distinct ID. In addition, there was an extra query happening on all event queries (fetching some person information) that wasn't needed and is now removed.

File: posthog/api/event.py [hunk 1]
```diff
@@ -62,14 +62,8 @@ class Meta:
...
         if hasattr(event, "person_properties"):
-            if event.person_properties:  # type: ignore
-                return event.person_properties.get("email", event.distinct_id)  # type: ignore
-            else:
-                return event.distinct_id
-        try:
-            return event.person.properties.get("email", event.distinct_id)
-        except:
-            return event.distinct_id
+            return event.person_properties  # type: ignore
```
Comment (mariusandra) [inline posthog/api/event.py:65]: I'd call this something else. `person_properties` kind of implies that you're getting `person.properties`, not an object `{ is_identified, distinct_id, properties: {} }`

**Review by mariusandra — COMMENTED** [review]
Codewise looks good. I have one nit regarding naming of one thing, but it's fine to merge without that.

The only UX question (regression?) I see is that now all anonymous users look like one and the same user:

image

If you only or mostly have anonymous users, the events list is less parsable now. Perhaps there's something we can do? E.g. a randomly generated color based on the uuid? Showing the last 12 symbols of the distinct id as "anonymous user af415a9438ec"?

**Review by Twixes — COMMENTED** [review]
My review was requested, but basically 👆 what Marius said

**Comment by paolodamico** [comment]
All feedback addressed! Thanks for the review @mariusandra! Decided to go with the last 5 characters as it seems to be significant enough without it becoming too intrusive, but we can always iterate. Decided also against the colors thing because I think we'll need at least 15-20 colors to have some significant indicator and it'll look like a rainbow in there.

**Pull request merged**